tests/components/metoffice/test_weather.py

```diff
@@ -35,13 +37,73 @@ async def test_site_cannot_connect(hass, requests_mock, legacy_patchable_time):
     await hass.config_entries.async_setup(entry.entry_id)
     await hass.async_block_till_done()
 
-    assert hass.states.get("weather.met_office_wavertree") is None
+    assert hass.states.get("weather.met_office_wavertree_3hourly") is None
+    assert hass.states.get("weather.met_office_wavertree_daily") is None
     for sensor_id in WAVERTREE_SENSOR_RESULTS:
         sensor_name, sensor_value = WAVERTREE_SENSOR_RESULTS[sensor_id]
         sensor = hass.states.get(f"sensor.wavertree_{sensor_name}")
         assert sensor is None
 
 
+@patch(
+    "datapoint.Forecast.datetime.datetime",
+    NewDateTime,
+)
+async def test_site_list_cannot_update(hass, requests_mock, legacy_patchable_time):
+    """Test we handle cannot connect error."""
+
+    # all metoffice test data encapsulated in here
+    mock_json = json.loads(load_fixture("metoffice.json"))
+    all_sites = json.dumps(mock_json["all_sites"])
+    wavertree_hourly = json.dumps(mock_json["wavertree_hourly"])
+    wavertree_daily = json.dumps(mock_json["wavertree_daily"])
+
+    requests_mock.get("/public/data/val/wxfcs/all/json/sitelist/", text=all_sites)
+    requests_mock.get(
+        "/public/data/val/wxfcs/all/json/354107?res=3hourly", text=wavertree_hourly
+    )
+    requests_mock.get(
+        "/public/data/val/wxfcs/all/json/354107?res=daily", text=wavertree_daily
+    )
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=METOFFICE_CONFIG_WAVERTREE,
+    )
+    entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    entity = hass.states.get("weather.met_office_wavertree_3_hourly")
+    assert entity
+
+    entity = hass.states.get("weather.met_office_wavertree_daily")
+    assert entity
+
+    requests_mock.get("/public/data/val/wxfcs/all/json/sitelist/", text="")
+    requests_mock.get(
+        "/public/data/val/wxfcs/all/json/354107?res=3hourly", text=wavertree_hourly
+    )
+    requests_mock.get(
+        "/public/data/val/wxfcs/all/json/354107?res=daily", text=wavertree_daily
+    )
+
+    # Make site list unavailable
+    data = hass.data[DOMAIN][entry.entry_id][METOFFICE_DATA]
+    await data.async_update_site()
+    await hass.async_block_till_done()
```

Comment:
This is a hack purely to add test coverage to an almost impossible code path. Unfortunately, I cannot get test environment to tun on my local machine so I kinda do it blindly. I'm actually tempted to just call "private" function `data._update_site()` instead because this one fails due to event loop constraints.